libs/frontend/feature-command/src/lib/page/add-page.ts

```diff
@@ -42,15 +43,16 @@ export const undoAddPage = (site: ISite, data: IAddPageData) => {
   const deleteCount = deleteComponentWithId(site, rootId)
   context.nextId -= deleteCount
 
-  // Set active page and selected component
-  setActivePage(site.editor, activePageRoute)
-  setSelectedComponent(site.editor, prevSelectedComponent)
-
   // Remove component tree expand state
   const { componentTreeExpandedItems } = site.editor ?? {}
   if (componentTreeExpandedItems) {
     delete componentTreeExpandedItems[rootId]
   }
+
   // Trigger page change event
   triggerEditorEvent(site, EditorEventName.OnPageChange)
+
+  // Set active page and selected component
+  setActivePage(site.editor, activePageRoute)
+  setSelectedComponent(site, prevSelectedComponent)
```

Comment:
Hi @sampullman 

The scroll to affected component feature was added to the `setSelectedComponent` helper, so commands where `setSelectedComponent` is used will automatically scroll to the affected component when applied.
But since commands may also be applied when `triggerEditorEvent` is called, I have to move `setActivePage` and `setSelectedComponent` down so that we can select on the correct component.
Although this simple solution worked, the downside is that users might see a quick flash between components/pages during redo/undo.

Reply:
Thanks, I'll keep that in mind! I think it's fine.